### DISCUSSION TITLE
Workaround for upstream issue in SlidingWindowRateLimiter

### DISCUSSION THREAD

**Discussion by lonix1** [App-vNext/Polly#2351] [Q&A]
Seeking some advice.

I wanted to use the sliding window rate limiter, but [RateLimiterRejectedException.RetryAfter](https://github.com/App-vNext/Polly/blob/ea15b525317d96af934ada7003b9ff3b6f6d52af/src/Polly.RateLimiting/RateLimiterRejectedException.cs#L75) is always null. After much digging I discovered that it's actually an [upstream issue](https://github.com/dotnet/aspnetcore/issues/52411).

Without that retry property, that limiter/strategy is not useful because the point of it is to control access and recover from errors.

There are some options for Polly users:
1. Wait for the bug to be fixed, but that may only be in v10 / November 2025
2. use fixed window instead
3. some workaround

Has someone encountered this issue before? Is there a workaround which does not entail reinventing the wheel (because in that case it's better to use (2) instead).

**Comment by lonix1** [comment]
If anyone who lands on this issue also needs that functionality, please upvote the [upstream bug report](https://github.com/dotnet/aspnetcore/issues/52411).

**Comment by martincostello** [comment]
Upstream issue is resolved in .NET 11 RC1.
